src/chocolatey/infrastructure.app/services/TemplateService.cs

```diff
@@ -188,5 +187,21 @@ public void generate_file_from_template(ChocolateyConfiguration configuration, T
             _fileSystem.create_directory_if_not_exists(_fileSystem.get_directory_name(fileLocation));
             _fileSystem.write_file(fileLocation, template, encoding);
         }
+
+        private string parse_template_path(ChocolateyConfiguration configuration)
+        {
+            if (!configuration.NewCommand.TemplateName.Equals("default"))
+            {
+                var rawRelativePath = _fileSystem.combine_paths(_fileSystem.get_current_directory(), configuration.NewCommand.TemplateName);
+                var fullPath = _fileSystem.get_full_path(rawRelativePath);
+
+                if (_fileSystem.directory_exists(fullPath))

```

Comment:
So the order of checks here is:
1. Check if templatename is a path that exists
1. Check if templatename exists in the template directory
1. Throw an error

I'd personally switch the order of one and two, so the templatename is checked in the template directory first, and only if it does not exist is the templatename checked if it is a path that exists. 

Say if you have a template called msi that you use, but you also have a directory called msi. Then, as this is currently setup, if you try to use the msi template, it will actually be using the msi directory as the template, instead of the installed msi template.